src/watchful.ts

```diff
@@ -120,12 +120,14 @@ export class Watchful extends Construct implements IWatchful {
     this.dash?.addWidgets(...widgets);
   }
 
-  public addAlarm(alarm: cloudwatch.Alarm) {
-    if (this.alarmTopic) {
-      alarm.addAlarmAction(new cloudwatch_actions.SnsAction(this.alarmTopic));
-    }
+  public addAlarm(alarm: cloudwatch.IAlarm) {
+    if (hasAlarmAction(alarm)) {
+      if (this.alarmTopic) {
+        alarm.addAlarmAction(new cloudwatch_actions.SnsAction(this.alarmTopic));
+      }
 
-    alarm.addAlarmAction(...this.alarmActions);
+      alarm.addAlarmAction(...this.alarmActions);
+    }
```

Comment:
why can't we create a new CompositeAlarm that wraps this `IAlarm` in the else case, as I had in my example above?

Reply:
That example code has the same problem, which is that we can't test for the existence of `addAlarm` action on an instance of IAlarm in order to make the decision to instantiate a new CompositeAlarm. Additionally CompositeAlarm _does_ have `addAlarmAction` so it would fail the test and you'd never create a new CompositeAlarm. The problem is I can't pass a CompositeAlarm into the method in the first place as it only allows Alarm.

Reply:
I feel like we're missing each other on something. Can you explain to me why we cannot do the following?
```ts
  public addAlarm(alarm: cloudwatch.IAlarm) {
    if (!hasAlarmAction(alarm)) {
      alarm = new cloudwatch.CompositeAlarm(this, `Created Alarm ${this.createdAlarmCount++}`, {
        alarmRule: cloudwatch.AlarmRule.fromAlarm(alarm, cloudwatch.AlarmState.ALARM),
      });
    }

    if (this.alarmTopic) {
      alarm.addAlarmAction(new cloudwatch_actions.SnsAction(this.alarmTopic));
    }
    alarm.addAlarmAction(...this.alarmActions);
  }
```

Reply:
The issue with that is the subsequent code in `addAlarm` works perfectly fine for the CompositeAlarm construct. CompositeAlarm is not missing `addAlarmAction` and so there's no need to do additional branching inside the method. The only problem is the original parameter type is too narrow to allow CompositeAlarm.

This problem can also be solved by doing something like 
```typescript
wf.addAlarm((myCompositeAlarm as unknown) as Alarm);
```
but I dislike casting.

Reply:
Sorry, I'm still confused. I understand that CompositeAlarm is not missing `addAlarmAction`. The way my suggestion works is that if `alarm` is an `Alarm` or `CompositeAlarm`, then the actions are added to `alarm` directly. If `alarm` is some other `IAlarm` that doesn't implement `addAlarmAction`, then it is wrapped in a `CompositeAlarm`, and the actions are added to the composite alarm.

Reply:
Ah okay, there's no such class at present, but it might be a good idea to guard against it existing at some point. The problem is TypeScript works against us here as we'd have to _again_ use `hasAlarmAction` to assert the alarm has that method. Otherwise, we're back to seeing it as IAlarm and no `addAlarmAction`. 👎🏻 

We'd end up with something like this
```typescript
  public addAlarm(alarm: cloudwatch.IAlarm) {
    if (!hasAlarmAction(alarm)) {
      alarm = new cloudwatch.CompositeAlarm(this, `Created Alarm ${this.createdAlarmCount++}`, {
        alarmRule: cloudwatch.AlarmRule.fromAlarm(alarm, cloudwatch.AlarmState.ALARM),
      });
    }
    if (hasAlarmAction(alarm)) {
      if (this.alarmTopic) {
        alarm.addAlarmAction(new cloudwatch_actions.SnsAction(this.alarmTopic));
      }

      alarm.addAlarmAction(...this.alarmActions);
    }
  }
```

You'd probably want to if/else it, but I can't think of a good way to do that.

Or maybe this is a little better?
```typescript
  public addAlarm(alarm: cloudwatch.IAlarm) {
    if (hasAlarmAction(alarm)) {
      if (this.alarmTopic) {
        alarm.addAlarmAction(new cloudwatch_actions.SnsAction(this.alarmTopic));
      }

      alarm.addAlarmAction(...this.alarmActions);
    } else {
      const ca = new cloudwatch.CompositeAlarm(this, `Created Alarm ${this.createdAlarmCount++}`, {
        alarmRule: cloudwatch.AlarmRule.fromAlarm(alarm, cloudwatch.AlarmState.ALARM),
      });
      if (this.alarmTopic) {
        ca.addAlarmAction(new cloudwatch_actions.SnsAction(this.alarmTopic));
      }

      ca.addAlarmAction(...this.alarmActions);
    }
  }
```

Might be able to rework it to try to avoid the repetition, but it definitely winds up being some extra LoC to cover the not existing use case. Do you think it's worth it?

Reply:
```ts
const alarmWithAction = hasAlarmAction(alarm) ? alarm :new cloudwatch.CompositeAlarm(this, `Created Alarm ${this.createdAlarmCount++}`, {
  alarmRule: cloudwatch.AlarmRule.fromAlarm(alarm, cloudwatch.AlarmState.ALARM),
});
if (this.alarmTopic) {
  alarmWithAction.addAlarmAction(new cloudwatch_actions.SnsAction(this.alarmTopic));
}
alarmWithAction.addAlarmAction(...this.alarmActions);
```

Reply:
Ah cool. That's more concise. Sorry, I didn't realize at first that you were trying to cover future cases. I've updated the PR.